connect/api/src/main/java/org/apache/kafka/connect/sink/SinkConnector.java

```diff
@@ -23,6 +23,9 @@
 
 /**
  * SinkConnectors implement the Connector interface to send Kafka data to another system.
+ * <p>Kafka Connect discovers extensions of this class using the Java {@link java.util.ServiceLoader} mechanism.
```

Comment:
Someone may see that their connector is loaded correctly by an older Connect worker and devalue this note because it appears to be inaccurate.

Perhaps something like "Newer versions of Kafka Connect discover...", or "Depending on the version and configuration of your Kafka Connect worker, extensions of this class may only be discoverable via the Java `ServiceLoader` mechanism"?